common/src/app/common/pages/changes.cljc

```diff
@@ -677,6 +677,26 @@
                           (map :id))]
           (into #{} xform shape-and-parents))))))
 
+(defmethod components-changed :mov-objects
+  [file-data {:keys [page-id _component-id parent-id] :as change}]
+  (when page-id
+    (let [page (ctpl/get-page file-data page-id)
+          all-parents (map #(ctn/get-shape page %)
+                           (into [parent-id] (cph/get-parent-ids (:objects page) parent-id)))
+          xform (comp (filter :main-instance?)
+                      (map :id))]
+      (into #{} xform all-parents))))
+
+(defmethod components-changed :del-obj
+  [file-data {:keys [id page-id _component-id] :as change}]
+  (when page-id
+    (let [page (ctpl/get-page file-data page-id)
+          shape-and-parents (map #(ctn/get-shape page %)
+                                 (into [id] (cph/get-parent-ids (:objects page) id)))
```

Comment:
Is more efficient here use `cons` instead of `into`

Reply:
done